packages/ember-htmlbars/tests/helpers/if_unless_test.js

```diff
@@ -1,260 +0,0 @@
-});
-
-QUnit.test('views within an if statement should be sane on re-render', function() {
-  view = EmberView.create({
-    template: compile('{{#if view.display}}{{view view.MyView}}{{/if}}'),
-    MyView: EmberView.extend({
-      tagName: 'input'
-    }),
-    display: false
-  });
-
-  runAppend(view);
-
-  equal(view.$('input').length, 0);
-
-  run(function() {
-    // Setting twice will trigger the observer twice, this is intentional
-    view.set('display', true);
-    view.set('display', 'yes');
-  });
-
-  var textfield = view.$('input');
-  equal(textfield.length, 1);
-
-  // Make sure the view is still registered in View.views
-  ok(EmberView.views[textfield.attr('id')]);
-});
-
-QUnit.test('the {{this}} helper should not fail on removal', function() {
-  view = EmberView.create({
-    context: 'abc',
-    template: compile('{{#if view.show}}{{this}}{{/if}}'),
-    show: true
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'abc', 'should start property - precond');
-
-  run(function() {
-    view.set('show', false);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('properties within an if statement should not fail on re-render', function() {
-  view = EmberView.create({
-    template: compile('{{#if view.value}}{{view.value}}{{/if}}'),
-    value: null
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.set('value', 'test');
-  });
-
-  equal(view.$().text(), 'test');
-
-  run(function() {
-    view.set('value', null);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('the {{this}} helper should not fail on removal', function() {
-  view = EmberView.create({
-    context: 'abc',
-    template: compile('{{#if view.show}}{{this}}{{/if}}'),
-    show: true
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'abc', 'should start property - precond');
-
-  run(function() {
-    view.set('show', false);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('edge case: child conditional should not render children if parent conditional becomes false', function() {
-  var childCreated = false;
-  var child = null;
-
-  view = EmberView.create({
-    cond1: true,
-    cond2: false,
-    viewClass: EmberView.extend({
-      init() {
-        this._super(...arguments);
-        childCreated = true;
-        child = this;
-      }
-    }),
-    template: compile('{{#if view.cond1}}{{#if view.cond2}}{{#view view.viewClass}}test{{/view}}{{/if}}{{/if}}')
-  });
-
-  runAppend(view);
-
-  ok(!childCreated, 'precondition');
-
-  run(function() {
-    // The order of these sets is important for the test
-    view.set('cond2', true);
-    view.set('cond1', false);
-  });
-
-  // TODO: Priority Queue, for now ensure correct result.
-  ok(!childCreated, 'child should not be created');
-  //ok(child.isDestroyed, 'child should be gone');
-  equal(view.$().text(), '');
-});
-
-QUnit.test('edge case: rerender appearance of inner virtual view', function() {
```

Comment:
deleted as this is covered by `Syntax test: {{#if}} and {{#unless}} without inverse` tests